Add "Special note for JVM"

Primetalk opened the pull request:
ref #341

Commit:
Add "Special note for JVM"

ref #341

haacked commented on index.md:
```diff
@@ -224,6 +224,19 @@ modification depends on whether you updated your dependencies in order to fix
 a bug or introduce new functionality. I would usually expect additional code
 for the latter instance, in which case it's obviously a minor level increment.
 
+#### Special note for JVM or similar environments
```
The FAQ is currently part of the Semver spec itself, thus this should go in https://github.com/semver/semver/blob/master/semver.md. When the next version of Semver is finalized, it'll get copied over here and translated.

Primetalk replied:
There is a PR https://github.com/semver/semver/pull/414 and a comment https://github.com/semver/semver/pull/414#pullrequestreview-80298503

Primetalk replied:
@Haacked There is an opposite opinion - https://github.com/semver/semver/pull/414#issuecomment-348630680. Could you discuss with @jwdonahue ?

jwdonahue replied:
@Primetalk, @Haacked is the authority on this, my apologies for any confusion.

fverdino commented on index.md:
```diff
@@ -224,6 +224,19 @@ modification depends on whether you updated your dependencies in order to fix
+
+In such environments where only one version of a library could 
+be present in a running system, the transitive dependencies effectively become
+part of the public library interface and thus the version of the user library SHOULD be 
```
Why not reword section in order to use MUST (instead of SHOULD) ?

It would be better to clarify when you MUST do something. Instead of simply recommending with SHOULD. 

Indeed, the potential effect is that libraries will still be semantic-versioning compliant even if they do not follow at all your recommendation! 

I suggest to add something like "[...] If any potential client environment allows only one version of a library be present in running system, [...] user library MUST be [...]"

Primetalk replied:
Updated

Commit:
Enforce MUST for version increments

In linearized environments where all dependencies are resolved to just a single version, we MUST increment the version of the library to the same extent as transitive dependencies change.

fverdino commented on index.md:
```diff
@@ -226,13 +226,15 @@ for the latter instance, in which case it's obviously a minor level increment.
 #### Special note for JVM or similar environments
 
-In such environments where only one version of a library could 
+I suggest to add something like "[...] If any potential client environment allows only one version of a library be present in running system, [...] user library MUST be [...]"
```
I guess line 229 (starting with "I suggest ...") was added in your last commit for mistake, it should be removed.

Primetalk replied:
Yeah, I'm sorry for the inconvenience.

Commit:
Remove "I suggest"

haacked commented:
Closing this until it's resolved on SemVer proper.

jwdonahue commented:
@Haacked, I thought we had resolved to do something like this in [#414](https://github.com/semver/semver/pull/414). Does this need to be a PR in semver/semver instead of here?
